test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,841 @@ TEST_F(SearchFilter, ExcludeRamp) {
+      close_single_edge(reader, tile, index, current, "JK", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    // TODO: 1->2 fails to get route without ignoring closure at 2. why?
+    auto result = gurka::route(closure_map, {"1", "3"}, costing,
+                               {{"/date_time/type", date_type},
+                                {"/date_time/value", "current"},
+                                {costing_speed_type, "current"}},
+                               reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EJ"});
+
+    // Specify search filter to disable exclude_closures at destination
+    const std::string& req_disable_exclude_closures =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s},{"lat":%s,"lon":%s,"search_filter":{"exclude_closures":false}}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("1").lat()) %
+         std::to_string(closure_map.nodes.at("1").lng()) %
+         std::to_string(closure_map.nodes.at("3").lat()) %
+         std::to_string(closure_map.nodes.at("3").lng()) % costing % costing % date_type)
+            .str();
+    result = gurka::route(closure_map, req_disable_exclude_closures, reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EJ", "JK"});
+  }
+}
+
+/*
+ *  Tests search_filter.exclude_closures at midway location
+ */
+TEST_P(ExcludeClosuresOnWaypoints, ExcludeClosuresAtMidway) {
+  std::string costing = std::get<0>(GetParam());
+  std::string date_type = std::get<1>(GetParam());
+  std::string costing_speed_type =
+      (boost::format("/costing_options/%s/speed_types/0") % costing).str();
+
+  // None of the edges are closed. Route has multiple waypoints
+  {
+    auto result = gurka::route(closure_map, {"1", "2", "3"}, costing,
+                               {{"/date_time/type", date_type},
+                                {"/date_time/value", "current"},
+                                {costing_speed_type, "current"}},
+                               reader);
+    gurka::assert::osrm::expect_steps(result, {"AB", "DE"});
+    // Note: Due to the way 2 is placed, it causes a uturn at E
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "DE", "EJ", "JK"});
+  }
+
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "DE", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    // TODO: 1->2 fails to get route. Why?
```

Comment:
Maybe what's happening is 2 snaps to E and therefore there's no route possible from 1->2?

```
    A--1--B---C----D-2-E--J--3--K
```

Reply:
Possibly. But its still failing after increasing length of `DE` and changing grid size to 100. Confusingly, the previous test case - `ExcludeClosuresAtDestination` no longer throws exception when routing between only 2 waypoints.

Reply:
I think we should debug this -- start by dumping out the geojson of the map using `dump_geojson(map)` and the input/snapped locations and map those in geojson.io.

Reply:
@danpaz so this route wasn't failing from `1->2`, but rather from `2->3` (which is expected since there are no suitable edges that can be used to route out of `2`). I'll update the comments to reflect this.